File: C/Exercice3_3.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char *argv[])
{
    char commande[256];

    if(argc == 1)
    {
        printf("Vous n'avez entrer aucun arguments\n");
        exit(0);
    }
    else if(argc >= 2)
    {
        snprintf(commande, sizeof(commande), "gedit %s", argv[1]);
        system(commande);
    }
    
    return 0;
}

File: Python/Exercice3_3.py
#coding: utf-8
import sys
import subprocess

if len(sys.argv) == 1:
    print("Vous n'avez entrer aucun arguments")
    sys.exit(0)
    
else:
    commande = f"gedit {sys.argv[1]}"
    subprocess.run(commande, shell=True)